Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=e3f47f89a57ef115755184a8b3f03a47ee227418
Commit:     e3f47f89a57ef115755184a8b3f03a47ee227418
Parent:     ede6f5aea054d3fb67c78857f7abdee602302043
Author:     Daniel Drake <dsd@gentoo.org>
AuthorDate: Wed Dec 10 23:28:25 2008 +0200
Committer:  Greg Kroah-Hartman <gregkh@suse.de>
CommitDate: Wed Jan 7 09:59:48 2009 -0800

    USB: unusual_devs.h additions for Pentax K10D
    
    Jaak Ristioja reported problems with his Pentax K10D camera:
    https://bugs.gentoo.org/show_bug.cgi?id=250406
    
    /proc/bus/usb/devices:
    T:  Bus=02 Lev=01 Prnt=01 Port=00 Cnt=01 Dev#=  2 Spd=12  MxCh= 0
    D:  Ver= 2.00 Cls=00(>ifc ) Sub=00 Prot=00 MxPS=64 #Cfgs=  1
    P:  Vendor=0a17 ProdID=006e Rev= 1.00
    S:  Manufacturer=PENTAX Corporation
    S:  Product=K10D
    C:* #Ifs= 1 Cfg#= 1 Atr=c0 MxPwr=  2mA
    I:* If#= 0 Alt= 0 #EPs= 3 Cls=08(stor.) Sub=06 Prot=50 Driver=usb-storage
    E:  Ad=81(I) Atr=02(Bulk) MxPS=  64 Ivl=0ms
    E:  Ad=02(O) Atr=02(Bulk) MxPS=  64 Ivl=0ms
    E:  Ad=83(I) Atr=03(Int.) MxPS=  64 Ivl=100ms
    The number of reported sectors is off-by-one.
    
    Signed-off-by: Daniel Drake <dsd@gentoo.org>
    Cc: Kadianakis George <desnacked@gmail.com>
    Cc: stable <stable@kernel.org>
    Signed-off-by: Phil Dibowitz <phil@ipom.com>

 drivers/usb/storage/unusual_devs.h |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/storage/unusual_devs.h b/drivers/usb/storage/unusual_devs.h
index bfcc1fe..a54212f 100644
--- a/drivers/usb/storage/unusual_devs.h
+++ b/drivers/usb/storage/unusual_devs.h
@@ -1425,6 +1425,13 @@ UNUSUAL_DEV( 0x0a17, 0x006, 0x0000, 0xffff,
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_FIX_INQUIRY ),
 
+/* Reported by Jaak Ristioja <Ristioja@gmail.com> */
+UNUSUAL_DEV( 0x0a17, 0x006e, 0x0100, 0x0100,
+		"Pentax",
+		"K10D",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_FIX_CAPACITY ),
+
 /* These are virtual windows driver CDs, which the zd1211rw driver
  * automatically converts into WLAN devices. */
 UNUSUAL_DEV( 0x0ace, 0x2011, 0x0101, 0x0101,
